### PR TITLE
COST-4764 Set lower level rbac roles when only top level are set for OpenShift cluster/node/project 

### PR DESCRIPTION
**Pull request by lcouzens**
## Jira Ticket

[COST-4764](https://issues.redhat.com/browse/COST-4764)

## Description

This change will fix partial matching for things like project/node filtering with a higher level rbac roles such as cluster by explicitly implying the correct lower level role accordingly. 

Example:

If you restrict to a single cluster this change will set node and project role to * (Access all items beneath that cluster)

## Testing

1. Checkout Main
2. run `python koku/manage.py shell`
3. `from koku.rbac import _apply_access`
4. ` _apply_access({'openshift.cluster': [{'operation': 'read', 'resources': ['test_cluster']}]})`
5. see returned access specifically the openshift sections just have test_cluster set
6. Checkout this branch and repeat steps 2-5 above
7. See the returned access now explicitly sets the lower lover access to `*` for node/project

Additional testing:
Try various access settings such as node and project levels:
`_apply_access({'openshift.cluster': [{'operation': 'read', 'resources': ['test_cluster']}], 'openshift.node': [{'operation': 'read', 'resources': ['test_node']}]})`
`_apply_access({'openshift.cluster': [{'operation': 'read', 'resources': ['test_cluster']}], 'openshift.node': [{'operation': 'read', 'resources': ['test_node']}], 'openshift.project': [{'operation': 'read', 'resources': ['test_project']}]})`

really adventurous you can try these too
`'openshift.node': [{'operation': 'read', 'resources': ['test_node']}]`
result:
'openshift.cluster': {'read': []},
 'openshift.node': {'read': ['test_node]},
 'openshift.project': {'read': ['*']},

`'openshift.project': [{'operation': 'read', 'resources': ['test_project']}]`
result:
'openshift.cluster': {'read': []},
 'openshift.node': {'read': []},
 'openshift.project': {'read': ['test_project']},

## Release Notes
- [x] proposed release note

```markdown
* [COST-4764](https://issues.redhat.com/browse/COST-4764) Fix Rbac issue around partial matching when filtering data
```

### CODE REVIEW

**Comment by maskarb** [comment]
What is being fixed? Should we be allowing partial matching when filtering? I don't think we should because this was explicitly disallowed with this issue:
https://issues.redhat.com/browse/COST-382

**Review by bacciotti — APPROVED** [review]

File: koku/koku/rbac.py [hunk 1]
```diff
@@ -106,6 +106,13 @@ def _update_access_obj(access, res_access, resource_list):
     """Update access object with access data."""
     for res in resource_list:
         access_items = access.get(res, [])
+        # Set OpenShift rbac inheritance cluster->node->project if higher level is restricted
+        if res == "openshift.node":
+            if not access_items and res_access["openshift.cluster"]["read"] != []:
+                access_items = [{"operation": "read", "resources": ["*"]}]
+        if res == "openshift.project":
+            if not access_items and res_access["openshift.node"]["read"] != []:
+                access_items = [{"operation": "read", "resources": ["*"]}]
```
Comment (maskarb) [inline koku/koku/rbac.py:115]: This does require the `resource_list` to be in a specific order. Maybe we should explicitly define the [`RESOURCE_TYPES`](https://github.com/project-koku/koku/blob/2fcd994ba04e69596bb8ebb2a81808ca029565be/koku/koku/rbac.py#L25) as an OrderedDict? I know key order is _now_ maintained in Python, but better safe than sorry?

Comment (lcouzens) [reply koku/koku/rbac.py:115]: Yeh I wondered this too tbh :+1:

**Review by maskarb — APPROVED** [review]
lgtm!

**Pull request merged**